// qubo/index.js
var root = '../../';
var paths = require(root + 'config/paths.json');

/**
 * Abstraction layer to load in different pieces of the applications
 * through a core module, without having to specify a full path in the
 * require() function
 */
module.exports = {
    app: function() {
        return this.getModule(paths.app);
    },
    pkg: function() {
        return this.getModule(paths.pkg);
    },
    routeConfig: function() {
        return this.getModule(paths.node.routes);
    },
    appConfig: function() {
        return this.getModule(paths.appConfig);
    },
    util: function(module) {
        return this.getModule(paths.node.util + '/' + module);
    },
    controller: function(module) {
        return this.getModule(paths.node.controllers + '/' + module);
    },
    model: function(module) {
        return this.getModule(paths.node.models + '/' + module);
    },
    /**
     * Returns an object of models from a given array of model name to load
     *
     * @param arr list of models to load in
     * @return obj key ()
     */
    models: function(arr) {
        var models = {},
            self = this;
        arr.forEach(function(modelName) {
            models[modelName] = self.model(modelName);
        });
        return models;
    },
    /**
     * Helper method to get modules from the qubo file directory
     *
     * @param src path from the root directory to module
     * @return module 
     */
    getModule: function(src) {
        return require(root + src);
    }
};
